### PR TITLE
NMS-15320: Added docs section for SameSite Cookie Attribute

### PR DESCRIPTION
**Pull request by christianpape**
* JIRA: https://opennms.atlassian.net/browse/NMS-15320

### CODE REVIEW

File: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
+
+= SameSite Cookie Attribute
+
+The SameSite Cookie Attribute was introduced to control access to cookies and helps to prevent various XSS- and CSRF-attacks.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc:4]:
```suggestion
The SameSite Cookie attribute controls access to cookies and helps prevent various cross-site scripting (XSS) and cross-site request forgery (CSRF) attacks.
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+This attribute tells browsers how to handle first- or third-party cookies and identifies whether or not to allow a cookie to be accessed.
+
+OpenNMS ships with the most secure configuration *strict* for its session cookie.
+This means, that the session cookie will not be sent for any cross-site request.
+There may be environments where a user want to relax this enforcement by setting the SameSite attribute to *lax*.
+
+== Relaxing the SameSite Attribute enforcement policy
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc:11]:
```suggestion
== Relax the SameSite attribute enforcement policy
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+In order to change the enforcement policy, you need to alter the file `$OPENNMS_HOME/jetty-webapps/opennms/WEB-INF/web.xml`.
```
Comment (Bonrob2) [inline docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc:13]:
```suggestion
To change the enforcement policy, you need to alter the `$OPENNMS_HOME/jetty-webapps/opennms/WEB-INF/web.xml` file.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
A few minor updates. Thank you!

File: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+[source, xml]
+----
+<session-config>
+  <cookie-config>
+    <http-only>true</http-only>
+    <comment>__SAME_SITE_STRICT__</comment> <1>
```
[the remark this answers is not in the document]
Comment (christianpape) [reply docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc:20]: Yes, the double undescores are needed.

**Review by Bonrob2 — DISMISSED** [review]
LGTM! Thank you!

File: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc [hunk 5]
```diff
@@ -0,0 +1,26 @@
+
+= SameSite Cookie Attribute
+
+The SameSite Cookie attribute controls access to cookies and helps prevent various cross-site scripting (XSS) and cross-site request forgery (CSRF) attacks.
+This attribute tells browsers how to handle first- or third-party cookies and identifies whether to allow a cookie to be accessed.
+
+OpenNMS ships with the most secure configuration, *strict*, for its session cookie.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc:7]:
```suggestion
{page-component-title} ships with the most secure configuration (`strict`) for its session cookie.
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+This means that the session cookie will not be sent for any cross-site request.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc:8]:
```suggestion
This means that the session cookie will not be sent for any cross-site requests.
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+There may be environments where a user wants to relax this enforcement by setting the SameSite attribute to *lax*.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc:9]:
```suggestion
There may be environments where a user wants to relax this enforcement by setting the SameSite attribute to `lax`.
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+
+== Relax the SameSite attribute enforcement policy
```
Comment (emarsh1995) [inline docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc:11]:
```suggestion
== Relax enforcement policy
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+
+To change the enforcement policy, you need to alter the `$OPENNMS_HOME/jetty-webapps/opennms/WEB-INF/web.xml` file.
```
Comment (emarsh1995) [inline docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc:13]:
```suggestion
To change the enforcement policy, you must alter `$OPENNMS_HOME/jetty-webapps/opennms/WEB-INF/web.xml`:
```

File: docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+
```
Comment (emarsh1995) [inline docs/modules/operation/pages/deep-dive/admin/configuration/same-site-cookie.adoc:14]:
```suggestion
+
```

**Review by emarsh1995 — CHANGES_REQUESTED** [review]

**Review by emarsh1995 — CHANGES_REQUESTED** [review]
One minor change; otherwise, LGTM. :)

**Review by Bonrob2 — APPROVED** [review]
LGTM!

**Pull request merged**